src/saturn_engine/utils/asyncutils.py

```diff
@@ -142,45 +142,72 @@ async def run(self) -> None:
 
 
 class DelayedThrottle(t.Generic[AsyncFNone]):
-    __call__: AsyncFNone
-
     def __init__(self, func: AsyncFNone, *, delay: float) -> None:
         self.func = func
         self.delay = delay
+        self.flush_event = asyncio.Event()
         self.delayed_task: t.Optional[asyncio.Task] = None
-        self.delayed_lock = asyncio.Lock()
```

Comment:
Isn't there a risk of race condition without the lock?

* `self.delayed_task = None` in `_delay_call` is executed just before `await asyncio.wait_for(self.delayed_task, timeout=timeout)` in `flush`
* `_call_` is called twice and we create the task twice
* ..?

Reply:
`self.delayed_task = None` is always called after the `await`, are you mixing it up with `self.delayed_params`?